lib/msf/core/handler/reverse_http.rb

```diff
@@ -84,6 +84,10 @@ def initialize(info = {})
         ),
         OptBool.new('IgnoreUnknownPayloads',
           'Whether to drop connections from payloads using unknown UUIDs'
+        ),
+        OptString.new('ShellcodeSize',
+          'The size of the shellcode be staged',
+          default: '0x400000'
```

Comment:
Consider:
```suggestion
        ),
        OptInt.new('ShellcodeSize',
          'The size of the shellcode be staged in Kilobytes',
          default: 4096
```

Calculations for for the rest of this PR would need to be adjusted. Accepting the numeric size in Kilobytes converted by the codebase seems more user friendly than expecting the user to convert the size to a hex value that would be directly consumed.  Conversion from int to hex by the codebase would also offer an opportunity for verification of boundary sizes and alignment.

Reply:
We can do but here I actually prefer grabbing the size in bytes myself - usually I can grab the exact number from whatever tool (xxd, stat etc)

Reply:
Using bytes makes sense to me, but I think I'd still be in favor of using `OptInt` here still for simplicity 🤔

```suggestion
        ),
        OptInt.new('ShellcodeSize',
          'The size of the shellcode be staged in bytes',
          default: 0x400000
```


> usually I can grab the exact number from whatever tool (xxd, stat etc)

Same, although I think I'd probably grab the byte count from `ls -la` or `wc -l` or similar. It also aligns with Window's VirtualAlloc `dwSize` etc.

Reply:
Makes sense 👍